pkg/controller/acmeorders/sync.go

```diff
@@ -473,6 +475,30 @@ func (c *controller) finalizeOrder(ctx context.Context, cl acmecl.Interface, o *
 		return fmt.Errorf("error finalizing order: %v", err)
 	}
 
+	if issuer.GetSpec().ACME != nil && issuer.GetSpec().ACME.PreferredChain != "" {
+		altBundles, err := cl.FetchCertAlternatives(ctx, certURL, true)
+		if err != nil {
+			return fmt.Errorf("error fetching alternate certificates: %w", err)
+		}
+		for _, altBundle := range altBundles {
+			for _, certPEM := range altBundle {
+				cert, err := x509.ParseCertificate(certPEM)
+				if err != nil {
+					return fmt.Errorf("error parsing alternate certificates: %w", err)
+				}
```

Comment:
Does this return the first or the last certificate in a multi-pem bundle?

Reply:
it returns all certs in the multi-pem bundle which were in the bundle of multi multi-pem bundles... gets confusing but that's how it works

Reply:
Ah sorry, I didn't read the code properly. Didn't see the nested `for` loop.

Reply:
I wondered if there was a helper function that could take a slice of certificates and construct a chain, with a leaf and a "root" (where root is the root of this chain rather than a global root cert).
It would return an error if the certificates in the bundle did not form a chain.
Then we could compare the CN to that  of the chain root.